cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/spark/SparkBatchSinkContext.java

```diff
@@ -51,4 +53,10 @@ public void addOutput(String datasetName, Map<String, String> arguments) {
   public void addOutput(String outputName, OutputFormatProvider outputFormatProvider) {
     sinkFactory.addOutput(getStageName(), outputName, outputFormatProvider);
   }
+
+  @Override
+  public void addOutput(Output output) {
+    Output trackableOutput = ExternalDatasets.makeTrackable(sparkContext.getAdmin(), output);
```

Comment:
Do we need to deprecate? Don't plugins specify explicitly what versions of the ETL apps they are compatible with? I believe that in 3.4 we are breaking compat anyway.